### PR TITLE
Reactor Netty HTTP Server Support

### PR DESCRIPTION
**Pull request by lalifk**
### Purpose

This PR introduces support for [Reactor Netty HTTP Server](https://projectreactor.io/docs/netty/release/reference/index.html#http-server).

**Note:**
Code to provide a Reactor Netty server adapter needed for these changes was already contributed to Resteasy by @crankydillo. See [resteasy#2693](https://github.com/resteasy/resteasy/pull/2693).

### High level summary of changes
There are 3 important changes to highlight:

#### **1.** New project structure

For clarity and also in order to re-use as much code as possible, a new project structure is proposed in this PR.

**From**
```
resteasy-spring-boot-starter-parent
				|
				|__ resteasy-spring-boot-starter
				|
				|__ spring-boot-sample-app
				|
				|__ spring-boot-sample-app-for-wildfly
				|
				|__ spring-boot-sample-app-no-jaxrs-application
				|
				|__ resteasy-spring-boot-starter-test
```

**To**

```
resteasy-spring-boot-starter-parent
				\__ resteasy-spring-boot-starter-common-parent
				:	|
				:	|__ resteasy-spring-boot-starter-common
				:	|
				:	|__ resteasy-spring-boot-starter-sample-app-common
				|
				\__ resteasy-servet-spring-boot-starter-parent
				:	|
				:	|__ resteasy-servlet-spring-boot-starter
				:	|
				:	|__ resteasy-servlet-spring-boot-starter-test
				:	|
				:	|__ resteasy-servlet-spring-boot-sample-app
				:	|
				:	|__ resteasy-servlet-spring-boot-sample-app-no-jaxrs
				:	|
				:	|__ resteasy-wildfly-spring-boot-sample-app
				|
				\__ resteasy-reactor-netty-spring-boot-starter-parent
				 	|
				 	|__ resteasy-reactor-netty-spring-boot-starter
				 	|
				 	|__ resteasy-reactor-netty-spring-boot-starter-test
				 	|
				 	|__ resteasy-reactor-netty-spring-boot-sample-app
				 	|
				 	|__ resteasy-reactor-netty-spring-boot-sample-app-no-jaxrs

```

####  **3.** READMEs updated to cover Reactor Netty HTTP server

### CODE REVIEW

**Comment by jamezp** [comment]
I've got no strong opinion as I really don't know a lot about Spring or Spring Boot. At a glance I did see some tabs that may have crept in we should likely remove :)

**Comment by lalifk** [comment]
> @lalifk Cool! In addition could you please fix the test error?
> 
I made another commit. Please try building again.

File: reactor-netty/resteasy-reactor-netty-spring-boot-starter/src/test/java/org/jboss/resteasy/springboot/JaxrsAppRegistrationTest.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+						TestApplication1.class,
+						TestApplication2.class)
+				.collect(Collectors.toList());		
+		
+		verifyExpectedApps(expectedApps, customPropertySource);
+
+	}
+	
+	
+	@Test
```
Comment (liweinan) [inline reactor-netty/resteasy-reactor-netty-spring-boot-starter/src/test/java/org/jboss/resteasy/springboot/JaxrsAppRegistrationTest.java:100]: needs some reformatting of the file

File: servlet/resteasy-servlet-spring-boot-sample-app-no-jaxrs/pom.xml [hunk 2]
```diff
@@ -6,15 +6,15 @@
 
     <parent>
         <groupId>org.jboss.resteasy</groupId>
-        <artifactId>resteasy-spring-boot-starter-parent</artifactId>
+        <artifactId>resteasy-servet-spring-boot-starter-parent</artifactId>
```
Comment (liweinan) [inline servlet/resteasy-servlet-spring-boot-sample-app-no-jaxrs/pom.xml:9]: here too

File: servlet/resteasy-servlet-spring-boot-sample-app/pom.xml [hunk 3]
```diff
@@ -5,17 +5,15 @@
 
     <parent>
         <groupId>org.jboss.resteasy</groupId>
-        <artifactId>resteasy-spring-boot-starter-parent</artifactId>
+        <artifactId>resteasy-servet-spring-boot-starter-parent</artifactId>
```
Comment (liweinan) [inline servlet/resteasy-servlet-spring-boot-sample-app/pom.xml:8]: I guess a global replacing would be fine.

File: servlet/pom.xml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>org.jboss.resteasy</groupId>
+        <artifactId>resteasy-spring-boot-starter-parent</artifactId>
+        <version>6.0.1-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>resteasy-servet-spring-boot-starter-parent</artifactId>
```
Comment (liweinan) [inline servlet/pom.xml:15]: misspelling

File: servlet/resteasy-servlet-spring-boot-starter-test/pom.xml [hunk 5]
```diff
@@ -5,15 +5,15 @@
 
     <parent>
         <groupId>org.jboss.resteasy</groupId>
-        <artifactId>resteasy-spring-boot-starter-parent</artifactId>
+        <artifactId>resteasy-servet-spring-boot-starter-parent</artifactId>
```
Comment (liweinan) [inline servlet/resteasy-servlet-spring-boot-starter-test/pom.xml:8]: same here

**Review by liweinan — CHANGES_REQUESTED** [review]

**Comment by asoldano** [comment]
> @lalifk In addition, we require all contributions to be made under the terms of the ASL 2.0 License:[ http://www.apache.org/licenses/LICENSE-2.0](http://www.apache.org/licenses/LICENSE-2.0)
> 
> Do you agree to these terms?

@liweinan , I believe @lalifk previously agreed with the license, see for example https://github.com/resteasy/Resteasy/pull/2242 .

**Comment by liweinan** [comment]
@asoldano okay!

**Comment by lalifk** [comment]
> @lalifk In addition, we require all contributions to be made under the terms of the ASL 2.0 License:[ http://www.apache.org/licenses/LICENSE-2.0](http://www.apache.org/licenses/LICENSE-2.0)
> 
> Do you agree to these terms?

I already did agree to those terms as pointed out by @asoldano.

**Comment by lalifk** [comment]
> Could you please do a cleanup? Thanks!
Cleanup complete!

**Pull request merged**